client/src/components/all-components/ALLCardFlip.js

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable no-unused-vars */
+import React, { useState } from "react";
+import PropTypes from "prop-types";
+import FlipIcon from "@mui/icons-material/Flip";
+
+const ALLCardFlip = (props) => {
+  const {
+    width,
+    height,
+    flipBack = false,
+    onAllFlipped,
+    frontImgURL,
+    cards,
+  } = props;
+  const [flipped, setFlipped] = useState(Array(cards.length).fill(false));
+
+  const flipCard = (index) => {
+    let newFlipped = flipped.slice();
+    if (flipBack) {
+      newFlipped[index] = !newFlipped[index];
+    } else if (!newFlipped[index]) {
+      newFlipped[index] = true;
+    } else {
+      return;
+    }
+
+    setFlipped(newFlipped);
+    if (newFlipped.filter((value) => value).length === cards.length) {
+      onAllFlipped();
+    }
+  };
+
+  return (
+    <div
+      className={`tw-w-full tw-grid tw-grid-cols-${width} tw-grid-rows-${height} tw-gap-4`}
+    >
+      {cards.map((card, i) => {
+        return (
+          <div
+            className={`${card.width == 2 ? "tw-col-span-2" : "tw-col-span-1"} tw-row-span-1 tw-perspective-distant`}
+            key={i}
+            onClick={() => flipCard(i)}
+          >
+            <div
+              className={`tw-relative tw-w-full tw-h-[15rem] tw-p-[0.2rem] tw-rounded-[1.2rem] tw-bg-gradient-to-r tw-from-labYellow tw-to-labBlue tw-duration-[0.5s] tw-transform-3d ${flipped[i] ? "tw-rotate-y-180" : ""}`}
+            >
```

Comment:
Clarifying question: are we not including a border around each of the cards?
This is what I see: ![Screenshot 2025-03-20 at 12 03 28 PM](https://github.com/user-attachments/assets/13d34723-3e61-4cfd-a69c-d50bb0b5dabc)

Figma: ![Screenshot 2025-03-20 at 12 02 58 PM](https://github.com/user-attachments/assets/5325ba08-4e7a-4a8a-95ef-52cd6ebfb9cd)